src/main/main.js

```diff
@@ -558,327 +556,142 @@ export const removePlayer (i){
-                                     mType[i],
-                                     true,
-                                     custcent[i].cs.x,
-                                     custcent[i].cs.y);
-    [player[i].inputs.rawlStickAxis[0].x,player[i].inputs.rawlStickAxis[0].y] =
-                  scaleToUnitAxes ( axisData(gamepad,"lsX"),
-                                    axisData(gamepad,"lsY"),
-                                    mType[i],
-                                    custcent[i].ls.x,
-                                    custcent[i].ls.y);
-    var lstickX = lsticks[0];
-    var lstickY = lsticks[1];
-    var cstickX = csticks[0];
-    var cstickY = csticks[1];
-
-    if (mType[i] == 3){
-      //console.log(gamepad.buttons[map.rA[mType[i]]]);
-      //-custcent[i].l
-      //-custcent[i].r
-      // FOR XBOX CONTROLLERS
-      var lAnalog = scaleToGCTrigger(buttonData(gamepad,"l").value, 0.2-custcent[i].l, 1); // shifted by +0.2
-      var rAnalog = scaleToGCTrigger(buttonData(gamepad,"r").value, 0.2-custcent[i].r, 1); // shifted by +0.2
-    }
-    else if (mType[i] == 2){
-      var lAnalog = scaleToGCTrigger(axisData(gamepad,"lA"),0.867-custcent[i].l, -0.6); // shifted by +0.867, flipped
-      var rAnalog = scaleToGCTrigger(axisData(gamepad,"rA"),0.867-custcent[i].r, -0.6); // shifted by +0.867, flipped
-    }
-    else if (mType[i] == 7) { //Brook adapter has no L/R analog information, just light presses
-      var lAnalog = gamepad.buttons[6].pressed ? 0.3 : 0;
-      var rAnalog = gamepad.buttons[7].pressed ? 0.3 : 0;
-    }
-    else {
-      var lAnalog = scaleToGCTrigger(axisData(gamepad,"lA"),0.867-custcent[i].l, 0.6); // shifted by +0.867
-      var rAnalog = scaleToGCTrigger(axisData(gamepad,"rA"),0.867-custcent[i].r, 0.6); // shifted by +0.867
-    }
+  let input = nullInputs;
+  
+  // the following block should be removed once the old input data is properly threaded through
+  for (var k = 1; k < 8; k++) {
+    input[k].lsX = player[i].inputs.lStickAxis[k-1].x;
+    input[k].lsY = player[i].inputs.lStickAxis[k-1].y;
+    input[k].rawX = player[i].inputs.rawlStickAxis[k-1].x;
+    input[k].rawY = player[i].inputs.rawlStickAxis[k-1].y;
+    input[k].csX = player[i].inputs.cStickAxis[k-1].x;
+    input[k].csY = player[i].inputs.cStickAxis[k-1].y;
+    input[k].lA = player[i].inputs.lAnalog[k-1];
+    input[k].rA = player[i].inputs.rAnalog[k-1];
+    input[k].s = player[i].inputs.s[k-1];
+    input[k].z = player[i].inputs.z[k-1];
+    input[k].a = player[i].inputs.a[k-1];
+    input[k].b = player[i].inputs.b[k-1];
+    input[k].x = player[i].inputs.x[k-1];
+    input[k].y = player[i].inputs.y[k-1];
+    input[k].r = player[i].inputs.r[k-1];
+    input[k].l = player[i].inputs.l[k-1];
+    input[k].dl = player[i].inputs.dpadleft[k-1];
+    input[k].dd = player[i].inputs.dpaddown[k-1];
+    input[k].dr = player[i].inputs.dpadright[k-1];
+    input[k].du = player[i].inputs.dpadup[k-1];
```

Comment:
This block needs to go, eventually, by passing the old input data to the `interpretInputs` function.